### PR TITLE
Model/employee

### PR DESCRIPTION
**Pull request by rutujanirmal**

### CODE REVIEW

File: db/migrate/20220920061443_create_employees.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateEmployees < ActiveRecord::Migration[6.0]
+  def change
+    create_table :employees do |t|
+      t.string :full_name, null: false
+      t.string :emp_id, null: false, unique: true
+      t.string :email
```
Comment (niharikabhoi2020) [inline db/migrate/20220920061443_create_employees.rb:6]: email should be present true and unique

File: db/migrate/20220920061443_create_employees.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      t.string :gender, null: false
+      t.integer :role , default: 0
```
Comment (niharikabhoi2020) [inline db/migrate/20220920061443_create_employees.rb:8]: use role as string. It will be easy to maintain and use

File: app/models/employee.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Employee < ApplicationRecord
+  validates :email , format: {with: EMAIL_REGEX}
+  validates :gender , format: {with: GENDER_REGEX}
+
+  enum role: {
+    "employee": 0,
+    "admin": 0
```
Comment (niharikabhoi2020) [inline app/models/employee.rb:7]: both keys have same value.

File: config/initializers/constants.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+EMAIL_REGEX = /^[a-z.]+\@joshsoftware\.(com|digital)$/i
+GENDER_REGEX = /^(M|F)$/
```
Comment (niharikabhoi2020) [inline config/initializers/constants.rb:2]: add new line at the end

**Review by niharikabhoi2020 — CHANGES_REQUESTED** [review]

**Comment by rutujanirmal** [comment]
Changed employee to user model after discussion and raised another PR for the same

**Pull request closed without merging**